knowledge_repo/app/migrations/versions/009eafe4838f_add_support_for_external_authentication.py

```diff
@@ -15,29 +15,28 @@
 
 
 def upgrade():
-    # Migrate from `usernames` to `identifiers`
     op.alter_column('users', 'username', new_column_name='identifier', existing_type=sa.String(length=500))
-    op.add_column('users', sa.Column('username', sa.String(length=500), nullable=True))
 
-    # Add new fields
-    op.add_column('users', sa.Column('password', sa.String(length=500), nullable=True))
-    op.add_column('users', sa.Column('active', sa.Boolean(), nullable=True))
-    op.add_column('users', sa.Column('name', sa.String(length=500), nullable=True))
-    op.add_column('users', sa.Column('preferred_name', sa.String(length=500), nullable=True))
-    op.add_column('users', sa.Column('avatar_uri', sa.Text(), nullable=True))
-    op.add_column('users', sa.Column('email', sa.String(length=500), nullable=True))
-    op.add_column('users', sa.Column('last_login_at', sa.DateTime(), nullable=True))
+    with op.batch_alter_table('users') as batch_op:
```

Comment:
For SQLite adding/removing of column constraints needs to occur within the `batch_alter_table` block. See [here](https://alembic.sqlalchemy.org/en/latest/ops.html#alembic.operations.Operations.batch_alter_table) for details.